ts/session/apis/open_group_api/sogsv3/sogsApiV3.ts

```diff
@@ -0,0 +1,501 @@
+ */
+async function handlePollInfoResponse(
+  statusCode: number,
+  pollInfoResponseBody: {
+    active_users: number;
+    read: boolean;
+    token: string;
+    upload: boolean;
+    write: boolean;
+    details: { admins?: Array<string>; image_id: number };
+  },
+  serverUrl: string,
+  roomIdsStillPolled: Set<string>
+) {
+  if (statusCode !== 200) {
+    window.log.info('handlePollInfoResponse subRequest status code is not 200');
+    return;
+  }
+
+  if (!isObject(pollInfoResponseBody)) {
+    window.log.info('handlePollInfoResponse pollInfoResponseBody is not object');
+    return;
+  }
+
+  const { active_users, read, upload, write, token, details } = pollInfoResponseBody;
+
+  if (!token || !serverUrl) {
+    window.log.info('handlePollInfoResponse token and serverUrl must be set');
+    return;
+  }
+
+  if (!roomIdsStillPolled.has(token)) {
+    window.log.info('handlePollInfoResponse room is no longer polled: ', token);
+    return;
+  }
+
+  const foundConvo = getSogsConvoOrReturnEarly(serverUrl, token);
+  if (!foundConvo) {
+    return; // we already print something in getSogsConvoOrReturnEarly
+  }
+  await foundConvo.setPollInfo({
+    read,
+    write,
+    upload,
+    subscriberCount: active_users,
+    details: pick(details, 'admins', 'image_id'),
+  });
+}
+
+async function filterOutMessagesInvalidSignature(
+  messagesFilteredBlindedIds: Array<OpenGroupMessageV4>
+) {
+  const sentToWorker = messagesFilteredBlindedIds.map(m => {
+    return {
+      sender: PubKey.cast(m.session_id).key, // we need to keep the prefix if this is a blinded or not pubkey
+      base64EncodedSignature: m.signature,
+      base64EncodedData: m.data,
+    };
+  });
+  const startVerify = Date.now();
+  const signatureValidEncodedData = (await callUtilsWorker(
+    'verifyAllSignatures',
+    sentToWorker
+  )) as Array<string>;
+  const signaturesValidMessages = compact(
+    (signatureValidEncodedData || []).map(validData =>
+      messagesFilteredBlindedIds.find(m => m.data === validData)
+    )
+  );
+  window.log.info(`[perf] verifyAllSignatures took ${Date.now() - startVerify}ms.`);
+
+  return signaturesValidMessages;
+}
+
+const handleSogsV3DeletedMessages = async (
+  messages: Array<OpenGroupMessageV4>,
+  serverUrl: string,
+  roomId: string
+) => {
+  const deletions = messages.filter(m => m.data === null);
+  const exceptDeletion = messages.filter(m => m.data !== null);
+  if (!deletions.length) {
+    return messages;
+  }
+  const allIdsRemoved = deletions.map(m => m.id);
+  try {
+    const convoId = getOpenGroupV2ConversationId(serverUrl, roomId);
+    const convo = getConversationController().get(convoId);
+    const messageIds = await Data.getMessageIdsFromServerIds(allIdsRemoved, convo.id);
+
+    await Promise.all(
+      (messageIds || []).map(async id => {
+        if (convo) {
+          await convo.removeMessage(id);
+        }
+        await Data.removeMessage(id);
+      })
+    );
+    //
```

Comment:
did you write something here before or should we delete?

Reply:
nope, just an empty comment line. Deleted :)